### PR TITLE
deliver only specified sample

### PR DESCRIPTION
**Pull request by senthil10**
Sometimes we wanted to only deliver one particular sample in a big project, currently we lack that. So this PR should fix it.

### CODE REVIEW

File: scilifelab/pm/core/deliver.py [hunk 1]
```diff
@@ -126,6 +126,13 @@ def raw_data(self):
         # Extract the list of samples and runs associated with the project and sort them
         samples = sorted(s_con.get_samples(fc_id=self.pargs.flowcell, sample_prj=self.pargs.project), key=lambda k: (k.get('project_sample_name','NA'), k.get('flowcell','NA'), k.get('lane','NA')))
 
+        # Deliver only mentioned samples if any
+        if self.pargs.sample:
+            samples = [sample_obj for sample_obj in samples if sample_obj.get('project_sample_name','NA') == self.pargs.sample]
+            if len(samples) == 0:
```
Comment (Galithil) [inline scilifelab/pm/core/deliver.py]: if not samples:

should work too, as empty lists evaluate to False, but it doesnt really matter

File: scilifelab/pm/core/deliver.py [hunk 2]
```diff
@@ -126,6 +126,13 @@ def raw_data(self):
...
+                self.log.error("There is no such sample {} for project {}".format(self.pargs.sample, self.pargs.project))
+                return
```
Comment (Galithil) [inline scilifelab/pm/core/deliver.py]: It's nice to explicit what you return, even if it's none

File: scilifelab/pm/core/deliver.py [hunk 3]
```diff
@@ -126,6 +126,13 @@ def raw_data(self):
         # Extract the list of samples and runs associated with the project and sort them
         samples = sorted(s_con.get_samples(fc_id=self.pargs.flowcell, sample_prj=self.pargs.project), key=lambda k: (k.get('project_sample_name','NA'), k.get('flowcell','NA'), k.get('lane','NA')))
 
+        # Deliver only mentioned samples if any
+        if self.pargs.sample:
+            samples = [sample_obj for sample_obj in samples if sample_obj.get('project_sample_name','NA') == self.pargs.sample]
```
Comment (Galithil) [inline scilifelab/pm/core/deliver.py]: it's more a filter than a list comprehension, but I guess this works too

File: scilifelab/pm/core/deliver.py (hunk 1 above)
Comment (senthil10) [inline scilifelab/pm/core/deliver.py]: hehe.. very true, that would be more pythonic :+1:

File: scilifelab/pm/core/deliver.py (hunk 2 above)
Comment (senthil10) [inline scilifelab/pm/core/deliver.py]: yeah, I was just trying to adapt the same style of the script :wink:

**Comment by senthil10** [comment]
@Galithil : I committed again for the `more pythonic` part, would be great if you can merge it :)

**Comment by Galithil** [comment]
Done !

**Comment by senthil10** [comment]
Awesome!! Tack @Galithil :)

**Pull request merged**